### ISSUE TITLE
Series should provide a "visitor" callback

### ISSUE THREAD

**Issue by chrisprice** [ScottLogic/d3-financial-components#202]
I want to be able to change the attributes of created elements on a per data-point basis.

Currently it is possible to change the class of a bar series element on a per data-point basis but not other attributes e.g. if I wanted to colour each bar programatically. This should be replaced with a "visitor" function of some kind (there might be a more appropriate d3 naming/mechanism for this I'm not sure) -

```
var bar = fc.series.bar()
        .xScale(dateScale)
        .yScale(priceScale)
        .yValue(fc.utilities.valueAccessor('close'))
        .visitor(function(d, s) {
            s.attr("stroke", "rgb(" + Math.floor(d.date.getDay() * 256 / 7) + ", 127, 127)");
        });
```

**Comment by chrisprice** [comment]
@ColinEberhardt has already created an implementation of this for his stacked bar example.

**Comment by ColinEberhardt** [comment]
closed by #229
